### PR TITLE
ENG-1770 dropdown typeahead component update + implementation

### PR DESCRIPTION
**Pull request by jeffgo10**
existing dropdown typeahead component is updated + additional implementation to group fields in content form & content filters (list and list modal)

### CODE REVIEW

**Comment by entando-jx** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **jeffgo10**

If they are not already assigned, you can assign the PR to them by writing `/assign @jeffgo10` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=entando%2Fentando-cms).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/entando/entando-cms/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by raxkaynan — CHANGES_REQUESTED** [review]
A few issues:
* The one about locale messages missing.
* Should the typeahead field support search by code only? I think it isn't user-friendly since the user would assume they would be able to search by the actual text that they see (e.g. typing 'access' should show 'Free Access'). No specific details about it have been mentioned in the ticket though. cc: @gidesan 
* The multi-select (e.g. Join groups) is now broken: 
image

File: src/ui/common/form/RenderDropdownTypeaheadInput.js
Comment (jeffgo10) [reply src/ui/common/form/RenderDropdownTypeaheadInput.js]: now amended i18n defs

**Comment by jeffgo10** [comment]
> A few issues:
> 
> * The one about locale messages missing.
> * Should the typeahead field support search by code only? I think it isn't user-friendly since the user would assume they would be able to search by the actual text that they see (e.g. typing 'access' should show 'Free Access'). No specific details about it have been mentioned in the ticket though. cc: @gidesan
> * The multi-select (e.g. Join groups) is now broken:
>   image

this could be an npm module installation issue. please clear

**Review by raxkaynan — APPROVED** [review]

**Comment by entando-jx** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/entando/entando-cms/pull/358#pullrequestreview-588702602" title="Approved">raxkaynan</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=entando%2Fentando-cms).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/entando/entando-cms/blob/master/OWNERS)~~ [raxkaynan]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
